core/src/main/java/bio/terra/pearl/core/service/participant/FamilyService.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.pearl.core.service.participant;
+
+import bio.terra.pearl.core.dao.participant.FamilyDao;
+import bio.terra.pearl.core.model.participant.Family;
+import bio.terra.pearl.core.service.CrudService;
+import bio.terra.pearl.core.service.exception.internal.InternalServerException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class FamilyService extends CrudService<Family, FamilyDao> {
+    public static final String FAMILY_SHORTCODE_ALLOWED_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    public static final int FAMILY_SHORTCODE_LENGTH = 6;
+    private final RandomUtilService randomUtilService;
+
+    public FamilyService(FamilyDao familyDao, RandomUtilService randomUtilService) {
+        super(familyDao);
+        this.randomUtilService = randomUtilService;
+    }
+
+    @Transactional
+    public Family create(Family family) {
+        if (family.getShortcode() == null) {
+            family.setShortcode(generateShortcode());
+        }
+        Family savedFamily = dao.create(family);
+        return savedFamily;
+    }
+
+    /**
+     * Generate a unique shortcode for a family. Operates exactly like the generateShortcode method in
+     * EnrolleeService.
+     */
```

Comment:
let's pull this into a utlity method, and also let's make sure that Enrollee and Family shortcodes have the same length and allowed chars.   I think the right approach is to have a ShortcodeUtils  class with the LENGTH and ALLOWED_CHARS utility method that takes as an argument a `shortcodeAlreadyExists` String -> boolean function.   

Should we prefix Family shortcodes with `F_` or some similar prefix so they're distinguishable at a glance from Enrollee shortcodes?